### DISCUSSION TITLE
Is there a way to enable guestFlush Parameter for GCP Backups

### DISCUSSION THREAD

**Discussion by azeemkhan0917** [vmware-tanzu/velero#5136] [Community support Q&A]
Is there a way to enable guestFlush Pramater for GCP Backups. I want to enable this Parameter in velero while backing up the resources 
https://cloud.google.com/compute/docs/reference/rest/v1/disks/createSnapshot?apix_params=%7B%22project%22%3A%22test%22%2C%22zone%22%3A%22test%22%2C%22disk%22%3A%22test%22%2C%22guestFlush%22%3Atrue%2C%22resource%22%3A%7B%7D%7D

<img width="648" alt="image" src="https://user-images.githubusercontent.com/83625641/180019885-fc2f54ee-d710-47e7-85cc-628a8fb83a71.png">

**Comment by kloyan** [comment]
I'm also interested in this. @azeemkhan0917 Did you find a way to use guest flush with the GCP plugin?
Looking at the [implementation](https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/main/velero-plugin-for-gcp/volume_snapshotter.go#L284), it might need to invoke [GuestFlush](https://pkg.go.dev/google.golang.org/api/compute/v1#DisksCreateSnapshotCall.GuestFlush) before executing the call.

**Comment by anshulahuja98** [comment]
SInce this is a VolumeSnapshot specific param. The current VolumeSnapshotter might not have inbuilt support for this. 
You can consider using the CSI Plugin for velero, and check if the GKE CSI driver supports this driver.

Setup: 
1. Validate if your PVCs are CSI PVCs
2. Validate if GKE CSI Driver supports this parameter, if not you can file an ask there.
3. Switch from using velero GKE VolumeSnapshotter plugin to velero CSI Plugin. 
4. Create a VolumeSnapshotClass with the above parameter.
